src/character.cpp

```diff
@@ -1563,6 +1575,18 @@ int Character::get_int_bonus() const
     return int_bonus;
 }
 
+int Character::ranged_dex_mod() const
+{
+    ///\EFFECT_DEX <20 increases ranged penalty
+    return std::max( ( 20 - get_dex() ) * 5, 0 );
```

Comment:
Dexterity is already better than perception in vast majority of cases, swapping that 5 and 2 between perception and dexterity would improve things.